provider/haproxy/config/haproxy_template.cfg

```diff
@@ -82,6 +82,6 @@ mode tcp
 {{else -}}
 mode {{$backend.Protocol}}
 {{end -}}
-{{range $j, $ep := $backend.Endpoints}}server {{$ep.Name}} {{$ep.IP}}:{{$ep.Port}} {{$ep.Config}}
+{{range $j, $ep := $backend.Endpoints}}server {{$ep.Name}} {{$ep.IP}}:{{$ep.Port}} {{ if $ep.Config }}{{$ep.Config}}{{else}} cookie {{ $j }}{{end}}
```

Comment:
Here is the example of how you can set the custom config on per server basis: 

https://docs.rancher.com/rancher/v1.2/en/cattle/adding-load-balancers/#custom-haproxy-configuration (refer to "server $IP" section)